pkg/operator/ceph/object/spec.go

```diff
@@ -318,37 +318,77 @@ func (c *clusterConfig) makeDaemonContainer(rgwConfig *rgwConfig) v1.Container {
 		return v1.Container{}
 	}
 	if kmsEnabled {
-		container.Args = append(container.Args,
-			cephconfig.NewFlag("rgw crypt s3 kms backend",
-				c.store.Spec.Security.KeyManagementService.ConnectionDetails[kms.Provider]),
-			cephconfig.NewFlag("rgw crypt vault addr",
-				c.store.Spec.Security.KeyManagementService.ConnectionDetails[api.EnvVaultAddress]),
-		)
-		if c.store.Spec.Security.KeyManagementService.IsTokenAuthEnabled() {
+		if len(c.store.Spec.SSEType) == 0 || strings.Contains(c.store.Spec.SSEType, "kms") {
 			container.Args = append(container.Args,
-				cephconfig.NewFlag("rgw crypt vault auth", kms.KMSTokenSecretNameKey),
-				cephconfig.NewFlag("rgw crypt vault token file",
-					path.Join(rgwVaultDirName, kms.VaultFileName)),
-				cephconfig.NewFlag("rgw crypt vault prefix", c.vaultPrefixRGW()),
-				cephconfig.NewFlag("rgw crypt vault secret engine",
-					c.store.Spec.Security.KeyManagementService.ConnectionDetails[kms.VaultSecretEngineKey]),
+				cephconfig.NewFlag("rgw crypt s3 kms backend",
+					c.store.Spec.Security.KeyManagementService.ConnectionDetails[kms.Provider]),
+				cephconfig.NewFlag("rgw crypt vault addr",
+					c.store.Spec.Security.KeyManagementService.ConnectionDetails[api.EnvVaultAddress]),
 			)
 		}
-		if c.store.Spec.Security.KeyManagementService.IsTLSEnabled() &&
-			c.clusterInfo.CephVersion.IsAtLeast(cephver.CephVersion{Major: 16, Minor: 2, Extra: 6}) {
+		if len(c.store.Spec.SSEType) != 0 && strings.Contains(c.store.Spec.SSEType, "s3") {
```

Comment:
Use a helper method instead of `strings.Contains(c.store.Spec.SSEType, "s3")`?  Something like `c.store.Spec.IsSSES3`?